tests/components/traccar/test_init.py

```diff
@@ -0,0 +1,226 @@
+        "accuracy": "10.5",
+        "batt": 10,
+        "speed": 100,
+        "bearing": "105.32",
+        "altitude": 102,
+    }
+
+    # Enter the Home
+    req = await traccar_client.post(url, params=data)
+    await hass.async_block_till_done()
+    assert req.status == HTTP_OK
+    state_name = hass.states.get(
+        "{}.{}".format(DEVICE_TRACKER_DOMAIN, data["id"])
+    ).state
+    assert STATE_HOME == state_name
+
+    # Enter Home again
+    req = await traccar_client.post(url, params=data)
+    await hass.async_block_till_done()
+    assert req.status == HTTP_OK
+    state_name = hass.states.get(
+        "{}.{}".format(DEVICE_TRACKER_DOMAIN, data["id"])
+    ).state
+    assert STATE_HOME == state_name
+
+    data["lon"] = 0
+    data["lat"] = 0
+
+    # Enter Somewhere else
+    req = await traccar_client.post(url, params=data)
+    await hass.async_block_till_done()
+    assert req.status == HTTP_OK
+    state_name = hass.states.get(
+        "{}.{}".format(DEVICE_TRACKER_DOMAIN, data["id"])
+    ).state
+    assert STATE_NOT_HOME == state_name
+
+    dev_reg = await hass.helpers.device_registry.async_get_registry()
+    assert len(dev_reg.devices) == 1
+
+    ent_reg = await hass.helpers.entity_registry.async_get_registry()
+    assert len(ent_reg.entities) == 1
+
+
+async def test_enter_with_attrs(hass, traccar_client, webhook_id):
+    """Test when additional attributes are present."""
+    url = "/api/webhook/{}".format(webhook_id)
+    data = {
+        "timestamp": 123456789,
+        "lat": "1.0",
+        "lon": "1.1",
+        "id": "123",
+        "accuracy": "10.5",
+        "batt": 10,
+        "speed": 100,
+        "bearing": "105.32",
+        "altitude": 102,
+    }
+
+    req = await traccar_client.post(url, params=data)
+    await hass.async_block_till_done()
+    assert req.status == HTTP_OK
+    state = hass.states.get("{}.{}".format(DEVICE_TRACKER_DOMAIN, data["id"]))
+    assert state.state == STATE_NOT_HOME
+    assert state.attributes["gps_accuracy"] == 10.5
+    assert state.attributes["battery_level"] == 10.0
+    assert state.attributes["speed"] == 100.0
+    assert state.attributes["bearing"] == 105.32
+    assert state.attributes["altitude"] == 102.0
+
+    data = {
+        "lat": str(HOME_LATITUDE),
+        "lon": str(HOME_LONGITUDE),
+        "id": "123",
+        "accuracy": 123,
+        "batt": 23,
+        "speed": 23,
+        "bearing": 123,
+        "altitude": 123,
+    }
+
+    req = await traccar_client.post(url, params=data)
+    await hass.async_block_till_done()
+    assert req.status == HTTP_OK
+    state = hass.states.get("{}.{}".format(DEVICE_TRACKER_DOMAIN, data["id"]))
+    assert state.state == STATE_HOME
+    assert state.attributes["gps_accuracy"] == 123
+    assert state.attributes["battery_level"] == 23
+    assert state.attributes["speed"] == 23
+    assert state.attributes["bearing"] == 123
+    assert state.attributes["altitude"] == 123
+
+
+@pytest.mark.xfail(
```

Comment:
Please add a test for two devices that update state:
https://github.com/home-assistant/home-assistant/blob/4349b8640e8bb480076698b7b20206ac9d99598e/tests/components/locative/test_init.py#L241-L278